source/dsl/GitDSL.ts

```diff
@@ -64,8 +76,7 @@ export interface GitDSL {
    * @param {string} filename the path to the json file
    * @param {string[]} diffTypes An array containing: "add", "del" or "normal" - then you can only get these changes
    */
-
-  diffForFile(filename: string, diffTypes?: string[]): string | null,
+  diffForFile(filename: string, diffTypes?: string[]): Promise<TextDiff | null>,
```

Comment:
Here's what I think would be more useful for most folks:

```ts
/** All Text diff values will be this shape */

/** All Text diff values will be this shape */
export interface TextDiff {
  /** The value before the PR's applied changes */
  before: string
  /** The value after the PR's applied changes */
  after: string,
  /** A string containing the full set of changes */
  diff: string,
  /** A string containing just the added lines */
  added: string,
  /** A string containing just the removed lines */
  removed: string,
}
```

This covers all cases except for the odd one where you only want `normal`/contextual lines, which seems quite fringe to me.